pkg/odo/cli/component/push.go

```diff
@@ -101,22 +101,40 @@ func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string)
 
 // Validate validates the push parameters
 func (po *PushOptions) Validate() (err error) {
-	return component.ValidateComponentCreateRequest(po.Context.Client, po.localConfig.GetComponentSettings(), false)
+	if err = component.ValidateComponentCreateRequest(po.Context.Client, po.localConfig.GetComponentSettings(), false); err != nil {
+		return err
+	}
+
+	isCmpExists, err := component.Exists(po.Context.Client, po.localConfig.GetName(), po.localConfig.GetApplication())
+	if err != nil {
+		return err
+	}
+
+	if !isCmpExists && po.pushSource && !po.pushConfig {
+		return fmt.Errorf("Component %s does not exist and hence cannot push only source. Please use `odo push` without any flags or with both `--source` and `--config` flags", po.localConfig.GetName())
+	}
+
+	return nil
 }
 
 func (po *PushOptions) createCmpIfNotExistsAndApplyCmpConfig(stdout io.Writer) error {
```

Comment:
this might be a good time to break this function into two` createCmpIfNotExists` and `applyCmdConfig`. there is alot of responsibility on a single function

Reply:
But applyconfig is just a single function call the implementation is already out in a different function